operator/controllers/clusterconfig_controller.go

```diff
@@ -47,15 +58,72 @@ type ClusterConfigReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.1/pkg/reconcile
 func (r *ClusterConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	reqLogger := log.WithValues("Request.Name", req.Name)
+	reqLogger.Info("reconciling ClusterConfig: START")
+	defer reqLogger.Info("reconciling ClusterConfig: STOP")
+
+	instance := &metalk8sscalitycomv1alpha1.ClusterConfig{}
+	err := r.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			if req.Name == instanceName {
+				panic(fmt.Errorf(
+					"%s ClusterConfig object should not be deleted", req.Name,
+				))
+			}
+			reqLogger.Info("ClusterConfig already deleted: nothing to do")
+			return endReconciliation(nil)
+		}
+		reqLogger.Error(err, "cannot read ClusterConfig: requeue")
+		return delayedRequeue(err)
+	}
+
+	if instance.Name != instanceName {
+		if err := r.Delete(ctx, instance); err != nil {
+			reqLogger.Error(
+				err, "cannot delete extra ClusterConfig: requeue",
+			)
+			return delayedRequeue(err)
+		}
+		reqLogger.Info("deleting extra ClusterConfig, consider updating the default one")
+		return endReconciliation(nil)
+	}
 
 	// TODO(user): your logic here
 
-	return ctrl.Result{}, nil
+	return endReconciliation(nil)
+}
+
+// Trigger a reschedule after a short delay.
+func delayedRequeue(err error) (ctrl.Result, error) {
+	delay := 10 * time.Second
+	return ctrl.Result{Requeue: err == nil, RequeueAfter: delay}, err
+}
+
+// Trigger a reschedule as soon as possible.
+func requeue(err error) (ctrl.Result, error) {
+	return ctrl.Result{Requeue: err == nil}, err
+}
+
+// Don't trigger a reschedule, we're done.
+func endReconciliation(err error) (ctrl.Result, error) {
+	return ctrl.Result{}, err
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *ClusterConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	client := mgr.GetClient()
+
+	instance := &metalk8sscalitycomv1alpha1.ClusterConfig{
+		ObjectMeta: metav1.ObjectMeta{Name: instanceName},
```

Comment:
Could be worth adding a finalizer, just to prevent mistakes (which could be quite troublesome if there were a lot of customizations in the object)

Reply:
I think we can cancel a deletion by removing `metadata.deletionTimestamp` and `metadata.deletionGracePeriodSeconds`, but yeah OK, since we will add the webhook I guess that's fine. And since we don't want to handle deletion, adding a finalizer doesn't make much sense.

For "resetting to defaults", I think updating the object to have `spec: null` could be enough, no need to delete.

Reply:
AFAIK, we cannot remove those fields.